org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/Id.java

```diff
@@ -0,0 +1,154 @@
+package org.eclipse.scanning.api.database;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * 
+ * This class represents one or more Ids as returned
+ * by an operation on the database. 
+ * 
+ * It is desirable to have operations which do more than
+ * one thing, therefore we allow compound operations.
+ * Accordingly there is more than one return code for operations
+ * like this. 
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class Id {
+
+	public static final Id NONE = new Id(0L);
+	
+	private Map<String, Long>   ids;
+	private Map<String, String> errors;
+	
+	public Id() {
+		ids    = new LinkedHashMap<>();
+		errors = new LinkedHashMap<>();
+	}
+	
+	public Id(Long id) {
+		this();
+		ids.put(null, id);
+	}
+	public Id(String name, Long id) {
+		this();
+		ids.put(name, id);
+	}
+	
+    /**
+     * 
+     * @return if this id contains one or more real ids
+     */
+	public boolean is() {
+		if (ids.isEmpty()) return false;
+		for (String key : ids.keySet()) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2864)